pkg/promotion/env_location.go

```diff
@@ -15,27 +14,35 @@ type EnvLocation struct {
 	Folder   string
 }
 
-func (env EnvLocation) IsLocal() bool {
+func (env EnvLocation) IsLocal() (bool, error) {
 	parsed, err := url.Parse(env.RepoPath)
 	if err != nil {
-		log.Printf("Error while parsing URL for environment path: '%v', assuming it is local.\n", err)
-		return true
+		return false, fmt.Errorf("failed parsing URL for environment path: %w", err)
 	}
-	return parsed.Scheme == ""
+	return parsed.Scheme == "", nil
 }
 
 // String implements Stringer so that messages (including logs, commits and PRs)
 // can reference EnvLocations consistently and in a more readable way.
+// When possible, it will shorten repository paths so that 'github.com/org/repo'
+// is just 'repo'. When not possible, it will use the entire URL.
 func (env EnvLocation) String() string {
-	if env.IsLocal() {
+	var repoName string
+	local, err := env.IsLocal()
```

Comment:
Better, this does feel a little odd (not due to your PR) that the way we detect it being a local repo is if we get an error parsing the URL (what if it is a remote repository but the user provided a typo somewhere - we'll go ahead assuming things are local, but maybe that doesn't matter).

Changes here look fine

Reply:
My changes might have made this less clear, but it's not just on Parse errors that we detect it as local. It's if the scheme comes back as `""`. Original snippet:
```
func fromLocalRepo(s string) bool {
	parsed, err := url.Parse(s)
	if err != nil || parsed.Scheme == "" {
		return true
	}
	return false
}
```

It could be good to revisit this and make it more robust, but I don't think it would be a priority. #104 could impact this anyway

Reply:
I see, yeah, I think this is fine